Checkstyle's configPath is erroneous

Hi there!

Removing the old checkstyle's config from my project made the build failing using the version `0.0.2-SNAPSHOT` of the plugin.

**What happened**

The `checkstyleIntegrationTest` was failing after the old configuration has been removed.
```
> Task :checkstyleIntegrationTest FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':checkstyleIntegrationTest'.
> Unable to create Root Module
```
Checking what's going on with `--stacktrace` shown me the underlying cause:

`Caused by: com.puppycrawl.tools.checkstyle.api.CheckstyleException: Unable to find: /home/fthouraud/Development/ekinow/api/gateway/file:/home/fthouraud/.gradle/caches/modules-2/files-2.1/com.ekino.oss.gradle.plugin/gradle-quality-plugin/0.0.2-SNAPSHOT/783f43c13cb521a9c535454a4d9804d8c6ec43be/gradle-quality-plugin-0.0.2-SNAPSHOT.jar!/checkstyle.xml`

The path of the resource has been prefixed by the absolute path of the project which can't work due to the resource path starting with `file` and already being absolute.

**What is expected**

We can expect that the checkstyle task could load its configuration from the plugin and run the analysis like intended.

It seems to work as intented before the `0.0.2-SNAPSHOT` but I can't tell which modification introduced this.

**Analysis**

I thought at first that was somewhat related to this [issue](https://github.com/gradle/gradle/issues/8286) but checking with both Java 8 and 11 gave me the same results.

To check this, I created a task to print some informations:
```
> Task :bug
*********************
Resource path: file:/home/fthouraud/.gradle/caches/modules-2/files-2.1/com.ekino.oss.gradle.plugin/gradle-quality-plugin/0.0.2-SNAPSHOT/783f43c13cb521a9c535454a4d9804d8c6ec43be/gradle-quality-plugin-0.0.2-SNAPSHOT.jar!/checkstyle.xml
As file: file:/home/fthouraud/.gradle/caches/modules-2/files-2.1/com.ekino.oss.gradle.plugin/gradle-quality-plugin/0.0.2-SNAPSHOT/783f43c13cb521a9c535454a4d9804d8c6ec43be/gradle-quality-plugin-0.0.2-SNAPSHOT.jar!/checkstyle.xml

Extension's configFile: /home/fthouraud/Development/ekinow/api/gateway/file:/home/fthouraud/.gradle/caches/modules-2/files-2.1/com.ekino.oss.gradle.plugin/gradle-quality-plugin/0.0.2-SNAPSHOT/783f43c13cb521a9c535454a4d9804d8c6ec43be/gradle-quality-plugin-0.0.2-SNAPSHOT.jar!/checkstyle.xml
*********************
```
This confirm that the path is biased on the `CheckstyleExtension` side.

**Possible solution**

Since Gradle 2.2, there's a `config` property on the `CheckstyleExtension` object which rely on a `TextResource` instead of a `File` ([Gradle 6.0 CheckstyleExtension](https://docs.gradle.org/current/javadoc/org/gradle/api/plugins/quality/CheckstyleExtension.html#setConfig-org.gradle.api.resources.TextResource-)).

We could then replace this part:
```kotlin
configure<CheckstyleExtension> {
  toolVersion = "8.24"
  configFile = getCheckstyleConfig()
}
```
with:
```kotlin
configure<CheckstyleExtension> {
  toolVersion = "8.24"
  config = resources.text.fromFile(getCheckstyleConfig())
}
```

If this seems okay to you I could propose a PR.

Thanks

--- a/src/main/kotlin/com/ekino/oss/gradle/plugin/quality/QualityPlugin.kt
+++ b/src/main/kotlin/com/ekino/oss/gradle/plugin/quality/QualityPlugin.kt
@@ -17,7 +17,6 @@ import org.gradle.testing.jacoco.plugins.JacocoPlugin
 import org.gradle.testing.jacoco.tasks.JacocoReport
 import org.sonarqube.gradle.SonarQubeExtension
 import org.sonarqube.gradle.SonarQubePlugin
-import java.io.File
 
 class QualityPlugin: Plugin<Project> {
 
@@ -38,7 +37,7 @@ class QualityPlugin: Plugin<Project> {
         // Checkstyle configuration
         configure<CheckstyleExtension> {
           toolVersion = "8.24"
-          configFile = getCheckstyleConfig()
+          config = resources.text.fromString(getCheckstyleConfig())
         }
 
         // Jacoco configuration
@@ -99,9 +98,9 @@ class QualityPlugin: Plugin<Project> {
     }
   }
 
-  internal fun getCheckstyleConfig(): File = getFilePath()?.let { File(it) }
+  internal fun getCheckstyleConfig() = getFilePath()?.use { String(it.readAllBytes()) }
           ?: throw MissingResourceException("The checkstyle config file cannot be found")
 
-  internal fun getFilePath() = QualityPlugin::class.java.classLoader?.getResource("checkstyle.xml")?.path
+  internal fun getFilePath() = QualityPlugin::class.java.classLoader?.getResourceAsStream("checkstyle.xml")
 
 }

--- a/src/test/resources/project_with_test_and_integration_test/build.gradle
+++ b/src/test/resources/project_with_test_and_integration_test/build.gradle
@@ -1,6 +1,6 @@
 plugins {
     id 'com.ekino.oss.gradle.plugin.quality'
-    id 'org.unbroken-dome.test-sets' version '2.1.0'
+    id 'org.unbroken-dome.test-sets' version '2.2.1'
 }
 
 dependencies {

